### PR TITLE
Bump haskell.nix and avoid repeat projects in nix config.

### PR DESCRIPTION
**Pull request by jbgi**
bump of haskell.nix should also improve eval perf.

### CODE REVIEW

File: nix/haskell.nix [hunk 1]
```diff
@@ -37,28 +45,12 @@ let
...
-        packages.ouroboros-consensus.configureFlags = [ "--ghc-option=-Werror" ];
-        packages.ouroboros-consensus-byron.configureFlags = [ "--ghc-option=-Werror" ];
-        packages.ouroboros-consensus-byron-test.configureFlags = [ "--ghc-option=-Werror" ];
-        packages.ouroboros-consensus-byronspec.configureFlags = [ "--ghc-option=-Werror" ];
-        packages.ouroboros-consensus-cardano.configureFlags = [ "--ghc-option=-Werror" ];
-        packages.ouroboros-consensus-mock.configureFlags = [ "--ghc-option=-Werror" ];
-        packages.ouroboros-consensus-shelley.configureFlags = [ "--ghc-option=-Werror" ];
-        packages.ouroboros-consensus-shelley-test.configureFlags = [ "--ghc-option=-Werror" ];
-        packages.ouroboros-consensus-test-infra.configureFlags = [ "--ghc-option=-Werror" ];
-        packages.cardano-client.configureFlags = [ "--ghc-option=-Werror" ];
```
Comment (mrBliss) [inline nix/haskell.nix:61]: Nice :+1:

File: nix/haskell.nix [hunk 2]
```diff
@@ -14,11 +14,19 @@
 }:
 let
 
+  src = haskell-nix.haskellLib.cleanGit {
+      name = "ouroboros-network-src";
+      src = ../.;
+  };
+
+  projectPackages = lib.attrNames (haskell-nix.haskellLib.selectProjectPackages
+    (haskell-nix.cabalProject { inherit src; }));
```
Comment (rvl) [inline nix/haskell.nix:23]: Thanks @hamishmack

File: nix/haskell.nix [hunk 3]
```diff
@@ -37,28 +45,12 @@ let
           # "stm" "terminfo"
         ];
       }
+      { packages = (lib.listToAttrs (map (p: lib.nameValuePair
+        # Compile all local packages with -Werror:
+        p { configureFlags = [ "--ghc-option=-Werror" ]; }
+      ) projectPackages));}
```
Comment (rvl) [inline nix/haskell.nix:51]: I find this way easier to read:

```suggestion
      # Compile all local packages with -Werror:
      { packages = lib.genAttrs (name: { configureFlags = [ "--ghc-option=-Werror" ]; }); }
```

**Pull request merged**